examples/audio_recorder/test/tests.ts

```diff
@@ -1,155 +1,115 @@
 import { ActorSubclass } from '@dfinity/agent';
-import { ok, Test } from 'azle/test';
+import { jsonStringify } from 'azle';
+import { createTestResult, equals, Test } from 'azle/test';
 
 import { _SERVICE } from './dfx_generated/audio_recorder/audio_recorder.did';
 
 // TODO to be more thorough we could test all of the error cases as well
 
-let global_user: any;
-let global_recording: any;
+type Context = {
+    user: any;
+    recording: any;
+};
 
 export function get_tests(
     audio_recorder_canister: ActorSubclass<_SERVICE>
-): Test[] {
+): Test<Context>[] {
     return [
         {
             name: 'create_user',
-            test: async () => {
+            test: async (context) => {
                 const user =
                     await audio_recorder_canister.createUser('lastmjs');
 
-                global_user = user;
-
-                return {
-                    Ok:
-                        user.username === 'lastmjs' &&
-                        user.recordingIds.length === 0
-                };
+                const expectedUsername = 'lastmjs';
+                const expectedRecordingCount = 0;
+
+                return createTestResult<Context>(
```

Comment:
Easy, just shove all of the values into a tuple
```javascript
return equals<Context>(
    [user.username, user.recordingIds.length],
    [expectedUsername, expectedRecordingCount],
    { context: { ...context, user } }
);
```
This will maintain the majority of the information we got from the custom error message (and all of the information that is being tests) and it doesn't require any changes to any code. 

Anything that doesn't follow this patter shouldn't be using an equal method anyways.

And equals should be renamed to testEquality. 

Oh yes, and then we can get rid of that customEqual business and rename that to just equals

Reply:
Here are some failed attempts for posterity
```javascript
type Equal = {
    actual: any;
    expected: any;
    func?: (actual: any, expected: any) => boolean;
};

type MultiEqualsOptions<Context> = {
    failMessage?: string;
    toString?: (value: any) => string;
    context?: Context;
};

export function multiEquals<Context>(
    tests: Equal[],
    options?: MultiEqualsOptions<Context>
): AzleResult<string, Context> {
    const valueToString = options?.toString ?? jsonStringify;
    for (const { func, actual, expected } of tests) {
        const equals = func ?? deepEqual;
        if (equals(actual, expected)) {
            continue;
        } else {
            const message =
                options?.failMessage ??
                `Expected: ${valueToString(
                    expected
                )}, Received: ${valueToString(actual)}`;
            return fail(message);
        }
    }
    return succeed(options?.context);
}

export function otherMultiEquals<A, E, Context>(
    actual: A | any[],
    expected: E | any[],
    options?: EqualsOptions<Context> | EqualsOptions<Context>[]
): AzleResult<string, Context> {
    const equals = options?.customEquals ?? deepEqual;
    const valueToString = options?.toString ?? jsonStringify;

    if (equals(actual, expected)) {
        return succeed(options?.context);
    } else {
        const message =
            options?.errMessage ??
            `Expected: ${valueToString(expected)}, Received: ${valueToString(
                actual
            )}`;
        return fail(message);
    }
}
```